app/assets/javascripts/discourse/app/models/login-method.js

```diff
@@ -8,7 +8,7 @@ import Site from "discourse/models/site";
 import { i18n } from "discourse-i18n";
 
 export default class LoginMethod extends EmberObject {
-  static buildPostForm(url) {
+  static buildPostForm(url, params = {}) {
```

Comment:
Pass all the `params` using the "hidden" form instead of using query params & the form.